services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/EnhancedClientUtils.java

```diff
@@ -68,10 +69,12 @@ public static String cleanAttributeName(String key) {
         }
 
         if (dynamoDbEnhancedClientExtension != null) {
-            ReadModification readModification = dynamoDbEnhancedClientExtension.afterRead(itemMap,
-                                                                                          operationContext,
-                                                                                          tableSchema.tableMetadata());
-
+            ReadModification readModification = dynamoDbEnhancedClientExtension.afterRead(
+                DefaultDynamoDbExtensionContext.builder()
```

Comment:
I think if we're going to go public on this interface we should either implement it in public (make it a class and not an interface that includes a builder), or give them a static builder, in which case the Builder needs to go on the public interface too.

Reply:
as discussed people dont need to create these, just consume them, so it's fine I think